src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -389,18 +413,40 @@ class LocalLambdaRunner {
         }
     }
 
-    private async getEvent(): Promise<{}> {
+    private async getConfig(): Promise<HandlerConfig> {
         const workspaceFolder = vscode.workspace.getWorkspaceFolder(this.localInvokeArgs.document.uri)
         if (!workspaceFolder) {
-            return {}
+            return buildHandlerConfig()
         }
 
         const config: HandlerConfig = await getLocalLambdaConfiguration(
             workspaceFolder,
             this.localInvokeArgs.handlerName
         )
 
-        return config.event
+        return config
+    }
+
+    private getEnvironmentVariables(config: HandlerConfig): EnvironmentVariables {
+        if (!!config.environmentVariables) {
+            return {
+                [LocalLambdaRunner.TEMPLATE_RESOURCE_NAME]: config.environmentVariables
+            }
+        } else {
+            return {}
+        }
+    }
+
+    private async getResourceForHandlerFromTemplate(handler: string, lambdas: LocalLambda[]):
+        Promise<CloudFormation.Resource | undefined> {
+
+        for (const lambda of lambdas) {
```

Comment:
```typescript
return lambdas.find(lambda => lambda.handler === handler)
```

`Array.find` returns `undefined` if not item matches the predicate.

This method also doesn't need to be `async`. Consider removing it entirely and calling `lambdas.find(lambda => lambda.handler === handler)` inline.